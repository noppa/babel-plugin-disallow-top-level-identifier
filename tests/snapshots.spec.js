const babel = require('babel-core');
const fs = require('fs');
const path = require('path');
const {promisify} = require('util');

/* globals __dirname */
const testcasesDir = (...pathnames) => path.join(__dirname, 'testcases', ...pathnames);
const readFile = promisify(fs.readFile);

fs.readdirSync(testcasesDir())
  .filter(_ => path.extname(_) === '.js')
  .sort()
  .map(_ => ({
    filename: _,
    testName: _.match(/[0-9]+_([a-zA-Z-]+)/)[1].replace(/-/g, ' ')
  }))
  .forEach(({filename, testName}) => {

    it(`Should work with ${testName}`, async () => {
      const file = await readFile(testcasesDir(filename), 'utf8');
      const {code} = babel.transform(file);
      
      expect(code).toMatchSnapshot();
    });

  });

