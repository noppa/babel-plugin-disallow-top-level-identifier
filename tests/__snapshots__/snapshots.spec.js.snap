// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should work with simple usage 1`] = `
"const module = 'oo';
const foo = 'f' + module;

export { module, foo };"
`;
